src/lib/protocols/hl7.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * hl7.c
+ *
+ * Health Level Seven (HL7)
+ * 
+ * Copyright (C) 2023 - ntop.org
+ * Copyright (C) 2023 - V.G <jacendi@protonmail.com>
+ *
+ * This file is part of nDPI, an open source deep packet inspection
+ * library based on the OpenDPI and PACE technology by ipoque GmbH
+ *
+ * nDPI is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * nDPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with nDPI.  If not, see <http://www.gnu.org/licenses/>.
+ * 
+ */
+
+#include "ndpi_protocol_ids.h"
+
+#define NDPI_CURRENT_PROTO NDPI_PROTOCOL_HL7
+
+#include "ndpi_api.h"
+#include "ndpi_private.h"
+
+static void ndpi_search_hl7(struct ndpi_detection_module_struct *ndpi_struct,
+                            struct ndpi_flow_struct *flow)
+{
+  struct ndpi_packet_struct const * const packet = &ndpi_struct->packet;
+
+  NDPI_LOG_DBG(ndpi_struct, "search HL7\n");
+
+  if (((flow->detected_protocol_stack[0] == NDPI_PROTOCOL_HTTP) ||
+      (flow->detected_protocol_stack[1] == NDPI_PROTOCOL_HTTP)) &&
+      packet->content_line.ptr != NULL)
```

Comment:
Two different ifs
```
  if (((flow->detected_protocol_stack[0] == NDPI_PROTOCOL_HTTP) ||
      (flow->detected_protocol_stack[1] == NDPI_PROTOCOL_HTTP)) {
    if(packet->content_line.ptr != NULL) {
       ...
    }
    return;
}
```
The goal is to return in any case if we are in the "HTTP-subprotocol" data path